lms/validation/_api.py

```diff
@@ -132,3 +137,31 @@ def validate(self, data, **_kwargs):
             raise ValidationError(
                 "both bookID and cfi are mandatory with type is 'vitalsource'"
             )
+
+
+class APISyncBlackboardSchema(PyramidRequestSchema):
+    class Data(Schema):
+        class LMS(Schema):
+            tool_consumer_instance_guid = fields.Str(required=True)
+
+        class Course(Schema):
+            context_id = fields.Str(required=True)
+
+        class Assignment(Schema):
+            resource_link_id = fields.Str(required=True)
+
+        class GroupInfo(Schema):
+            class Meta:
+                unknown = INCLUDE
+
+        lms = fields.Nested(LMS, required=True)
+        course = fields.Nested(Course, required=True)
+        assignment = fields.Nested(Assignment, required=True)
+        group_info = fields.Nested(GroupInfo, required=True)
+
+        gradingStudentId = fields.Str(required=False, allow_none=True)
+
+    authUrl = fields.URL(required=True)
+    path = fields.Str(required=True)
+
+    data = fields.Nested(Data, required=True)
```

Comment:
The schema looks odd to me: I don't think that `authUrl` and `path` are part of the request body that the frontend posts to the backend. And the contents of `data` are just the top-level contents of the request body, they're not in a `data` subdict. Won't this result in a validation error when the frontend posts to the sync API? As I don't think the validation schema matches the shape of the request